src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Resource/FindAzureResourceCmdlet.cs

```diff
@@ -0,0 +1,492 @@
+
+        /// <summary>
+        /// The list tenant resources parameter set.
+        /// </summary>
+        internal const string ListTenantResourcesParameterSet = "Lists the resources based on the specified scope at the tenant level.";
+
+        /// <summary>
+        /// The get tenant resource parameter set.
+        /// </summary>
+        internal const string MultiSubscriptionListResourcesParameterSet = "Get a resources using a multi-subscription query.";
+
+        /// <summary>
+        /// Caches the current subscription ids to get all subscription ids in the pipeline.
+        /// </summary>
+        private readonly List<Guid> subscriptionIds = new List<Guid>();
+
+        /// <summary>
+        /// Gets or sets the resource name parameter.
+        /// </summary>
+        [Alias("Name")]
+        [Parameter(ParameterSetName = FindAzureResourceCmdlet.ListResourcesParameterSet, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The resource name substring. e.g. if your resource name is testResource, you can specify test.")]
+        [Parameter(ParameterSetName = FindAzureResourceCmdlet.ListTenantResourcesParameterSet, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The resource name substring. e.g. if your resource name is testResource, you can specify test.")]
+        [Parameter(ParameterSetName = FindAzureResourceCmdlet.MultiSubscriptionListResourcesParameterSet, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The resource name substring. e.g. if your resource name is testResource, you can specify test.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceNameContains { get; set; }
+
+        /// <summary>
+        /// Gets or sets the resource type parameter.
+        /// </summary>
+        [Parameter(ParameterSetName = FindAzureResourceCmdlet.ListResourcesParameterSet, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The resource type. e.g. Microsoft.Sql/Servers/Databases.")]
+        [Parameter(ParameterSetName = FindAzureResourceCmdlet.ListTenantResourcesParameterSet, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The resource type. e.g. Microsoft.Sql/Servers/Databases.")]
+        [Parameter(ParameterSetName = FindAzureResourceCmdlet.MultiSubscriptionListResourcesParameterSet, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The resource type. e.g. Microsoft.Sql/Servers/Databases.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceType { get; set; }
+
+        /// <summary>
+        /// Gets or sets the extension resource type.
+        /// </summary>
+        [Parameter(ParameterSetName = FindAzureResourceCmdlet.ListResourcesParameterSet, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The extension resource type. e.g. Microsoft.Sql/Servers/Databases.")]
```

Comment:
Extension resource type is something that comes after the original/first resource type. For example: Microsoft.Sql/servers/testblahvivek23/databases/testdb2vivek123

I've added this example now, but this is something we'll need to fix across our cmdlets and make this more clearer in our help files. I'll send a separate PR for that.